shop-ui/src/main/java/com/maxshop/viewModel/ProductDetailsViewModel.kt

```diff
@@ -61,6 +66,27 @@ internal class ProductDetailsViewModel @Inject constructor(
         }
     }
 
+    fun addToBag() {
+        viewModelScope.launch {
+            _detailedProduct.value?.let {
+                try {
+                    addProductToBagUseCase.execute(
+                        detailedProductMapper.toBagProduct(
+                            it,
+                            if (it.colorsList == null) null
```

Comment:
It took me some to remember and understand what is going on here. And in my case I previously heard that you've decided to handle size and color pickers visibility by null value. However if I see the code for the very first time I will not understand what is going on. So please hide this handling inside data class. And maybe add a documentation why the method work in that manner

The same for sizes